### ISSUE TITLE
#[path = "."] fails with: Error writing files: io error: Failed to find module bar in "." None

### ISSUE THREAD

**Issue by Veetaha** [rust-lang/rustfmt#4334]
**Describe the bug**

Both of the following definitions confuse `rustfmt`
```rust
#[path = "."] 
mod foo { ... }
#[path = ""] 
mod foo { ... }
```
`cargo fmt` reports respectively:
```
Error writing files: io error: Failed to find module bar in "." None
```
```
Error writing files: io error: Failed to find module bar in "" None
```

However `cargo build` works just fine.

**To Reproduce**

Follow the instructions in README.md here: https://github.com/Veetaha/rustfmt-bugreport

**Expected behavior**

Rustfmt should finish without an error formatting all files, including those referenced via `#[path = "foo.bar.rs"]` attribute.

**Meta**

- rustfmt version: `rustfmt 1.4.16-stable (939e164 2020-06-11)`
- From where did you install rustfmt?: `rustup`
- How do you run rustfmt: `cargo fmt`

I use this code to generate a nested module structure for files generated in a flat directory. It works when compiling but
formatting these files with `cargo fmt` currently doesn't work. If you know any workarounds I will appreciate!

**Comment by ayazhafiz** [comment]
This was fixed in #4194, but hasn't been released yet (only as a part of https://github.com/rust-lang/rustfmt/releases/tag/v2.0.0-rc.2)

**Comment by calebcartwright** [comment]
Thanks for the report @Veetaha and confirmation @ayazhafiz. Closing
